tools/db_bench_tool.cc

```diff
@@ -4777,8 +4780,25 @@ class Benchmark {
         }
       }
 
-      size_t id = thread->rand.Next() % num_key_gens;
+      if (write_mode != SEQUENTIAL) {
+        id = thread->rand.Next() % num_key_gens;
+      } else {
+        // When doing a sequential load with multiple databases, load them in
+        // order rather than all at the same time to avoid:
+        // 1) long delays between flushing memtables
```

Comment:
So with this change, each DB will hold a range of the keys (no overlapping) instead of having the same key range.

Reply:
Great question, but with this change each db will have keys from 0 to $num. Prior to this change they had keys from 0 to $almost-num. And prior to this change it was extremely unlikely that $almost-num == $num. Where:
* $num is the value from --num
* $almost-num is a value close to $num

Each db has a separate KeyGenerator
https://github.com/facebook/rocksdb/blob/main/tools/db_bench_tool.cc#L4656

Which is used here:
https://github.com/facebook/rocksdb/blob/main/tools/db_bench_tool.cc#L4885